core/src/main/java/org/elasticsearch/ingest/IngestMetadata.java

```diff
@@ -86,20 +95,11 @@ public void writeTo(StreamOutput out) throws IOException {
 
     @Override
     public MetaData.Custom fromXContent(XContentParser parser) throws IOException {
-        ObjectParser<Void, Void> ingestMetaDataParser = new ObjectParser<>("ingest_metadata", null);
-
         Map<String, PipelineConfiguration> pipelines = new HashMap<>();
-        ingestMetaDataParser.declareField((parser1, aVoid, aVoid2) -> {
```

Comment:
+1 this is much nicer. I like the fact we can tell ObjectParser to just work with ArrayList. I kind of thought that a custom POJO was always needed...